### PR TITLE
feat: added table max width support

### PR DESCRIPTION
**Pull request by privateOmega**

### CODE REVIEW

File: src/helpers/xml-builder.js [hunk 1]
```diff
@@ -1430,17 +1431,31 @@ const buildTable = (vNode, attributes, docxDocumentInstance) => {
...
       const percentageValue = tableStyles.width.match(percentageRegex)[1];
       width = Math.round((percentageValue / 100) * attributes.maximumWidth);
     }
 
     // Choose the higher of the two widths
     modifiedAttributes.width =
-      width && minimumWidth ? Math.max(width, minimumWidth) : width || minimumWidth;
+      width && minimumWidth
+        ? Math.max(Math.min(width, maximumWidth), minimumWidth)
+        : Math.min(maximumWidth, width || minimumWidth || Number.POSITIVE_INFINITY);
```
Comment (amrita-syn) [inline src/helpers/xml-builder.js:1458]: I elaborated all conditions for clarity, can you please check if the existing code covers all the below.
```suggestion
      if ( width ) {
          modifiedAttributes.width = width;
          if ( minimumWidth ) modifiedAttributes.width = Math.max(modifiedAttributes.width, minimumWidth);
          if ( maximumWidth ) modifiedAttributes.width = Math.min(modifiedAttributes.width, maximumWidth);
      } else {
           if ( minimumWidth  || maximumWidth ) {
              // max overrides min if both are specified
              modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.max(attributes.maximumWidth, minimumWidth);
          }
          
          if ( !minimumWidth  && !maximumWidth ) {
              // neither is specified, let word auto-process it
              delete modifiedAttributes.width;
          }
      }
```

Comment (privateOmega) [reply src/helpers/xml-builder.js:1458]: @amrita-syn Whats with this line, 

modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.max(attributes.maximumWidth, minimumWidth); ? **Math.min(attributes.maximumWidth, maximumWidth)**

Comment (privateOmega) [reply src/helpers/xml-builder.js:1458]: `width = Math.min(width, attributes.maximumWidth)` at the end should handle every situation where computed width might be greater than page width.

Comment (charuthaB) [reply src/helpers/xml-builder.js:1458]: `modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.max(attributes.maximumWidth, minimumWidth);` 
will need to be 
`modifiedAttributes.width = Math.min(attributes.maximumWidth, maximumWidth) || Math.min(attributes.maximumWidth, minimumWidth);`

**Pull request merged**